crates/polars-core/src/chunked_array/ops/lambda.rs

```diff
@@ -292,10 +317,36 @@ impl LambdaExpression {
             LambdaExpression::StaticStr(v) => AnyValue::String(v),
             LambdaExpression::Variable(idx) => (*args[*idx]).into(),
             LambdaExpression::GreaterThan(left, right) => {
-                AnyValue::Boolean(left.eval_numeric::<T>(args) > right.eval_numeric::<T>(args))
+                let left = left.eval_numeric::<T>(args);
+                let right = right.eval_numeric::<T>(args);
+
+                // Special case for Int8/Int16 and Int32 when using Array[Byte]/Array[Short]
+                // For example: Array[Byte](77) < 5
+                match (&left, &right) {
```

Comment:
Are left and right used anywhere, that they need to be a reference?

Reply:
i think linter stated they are being consumed by the match expression otherwise. i'll look into it again.